### PR TITLE
Issue 1874

### LINKED ISSUE #1873: BOLED: Presetscreen: Einschalten von "Direct Ld" kann "To Part" ausschalten (s.u.)
Der Bug tritt erst auf, wenn man etwas in den Part geladen hat:

Variante 1:
- Dual-Preset laden
- "To Part" einschalten
- "Direct Ld" einschalten
- Mit dem Encoder ein/en Preset/Part auswählen/laden
- "Direct Ld" ausschalten
- "Direct Ld" einschalten
  => "To Part" geht aus

Variante 2:
- Dual-Preset laden
- "To Part" einschalten
- "Direct Ld" ausschalten
- Mit dem Encoder ein/en Preset/Part auswählen und mit Enter laden
- "Direct Ld" einschalten
  => "To Part" geht aus

### LINKED ISSUE #1874: BOLED: To Part Voice Group Selection not synced with normal Voice Group Selection
toggling load to part can lead to inconsistent part indications

### PR DESCRIPTION
**Pull request by schlenkibus**
closes #1873 
closes #1874

### CODE REVIEW

File: projects/playground/src/presets/PresetManager.cpp [hunk 1]
```diff
@@ -234,15 +235,16 @@ void PresetManager::doAutoLoadSelectedPreset()
...
+    FocusAndMode focusAndMode = hwui->getFocusAndMode();
 
     bool isPresetManagerActive = (focusAndMode.focus == UIFocus::Banks || focusAndMode.focus == UIFocus::Presets);
     bool isStoring = (focusAndMode.mode == UIMode::Store);
     bool isStoringPreset = isPresetManagerActive && isStoring;
 
     if(!isStoringPreset)
     {
-      scheduleAutoLoadSelectedPreset();
+      autoLoadPresetAccordingToLoadType();
```
Comment (hhoegelo) [inline projects/playground/src/presets/PresetManager.cpp:247]: before, the loading was scheduled, now it is done synchronously - is this a problem?

File: projects/playground/src/presets/PresetManager.cpp [hunk 2]
```diff
@@ -937,3 +939,37 @@ void PresetManager::scheduleLoadToPart(const Preset *preset, VoiceGroup loadFrom
...
+  {
+    if(auto selPreset = bank->getSelectedPreset())
+    {
+      auto directLoadSetting = Application::get().getSettings()->getSetting<DirectLoadSetting>();
+      auto loadToPartActive = Application::get().getHWUI()->isInLoadToPart();
+
+      switch(selPreset->getType())
+      {
+        case SoundType::Single:
+          eb->undoableLoadSelectedPreset(currentVoiceGroup);
```
Comment (hhoegelo) [inline projects/playground/src/presets/PresetManager.cpp:958]: Does this mean, that, if we are in the part II of a dual preset, (currentVoiceGroup == II), we would load the selected preset info vg II only?

Comment (schlenkibus) [reply projects/playground/src/presets/PresetManager.cpp:958]: yes, undoableLoadSelectedPreset calls either           `undoableLoadPresetIntoDualSound(preset, loadInto);` or `        undoableLoad(preset);` depending on the preset type of selected

File: projects/playground/src/presets/PresetManager.cpp (hunk 1 above)
Comment (schlenkibus) [reply projects/playground/src/presets/PresetManager.cpp:247]: did not feel like a problem on the dev pc, but i will rewrite it to schedule the loading instead

**Pull request merged**